src/interface/src/app/password-reset/password-reset.component.ts

```diff
@@ -0,0 +1,92 @@
+import { Component, NgZone, OnInit } from '@angular/core';
+import {
+  AbstractControl,
+  FormBuilder,
+  FormGroup,
+  Validators,
+} from '@angular/forms';
+import { ActivatedRoute, Data, Router } from '@angular/router';
+import { HttpErrorResponse } from '@angular/common/http';
+import { MatDialog } from '@angular/material/dialog';
+
+import { AuthService, PasswordResetToken } from '../services';
+import { ConfirmationDialogComponent } from './confirmation-dialog/confirmation-dialog.component';
+
+@Component({
+  selector: 'app-password-reset',
+  templateUrl: './password-reset.component.html',
+  styleUrls: ['./password-reset.component.scss'],
+})
+export class PasswordResetComponent implements OnInit {
+  errors: string[] = [];
+
+  form: FormGroup;
+
+  passwordResetToken: PasswordResetToken | null = null;
+
+  constructor(
+    private activatedRoute: ActivatedRoute,
+    private authService: AuthService,
+    private formBuilder: FormBuilder,
+    private router: Router,
+    private ngZone: NgZone,
+    private readonly dialog: MatDialog
+  ) {
+    this.form = this.formBuilder.group(
+      {
+        password1: this.formBuilder.control('', [
+          Validators.required,
+          Validators.minLength(8),
+        ]),
+        password2: this.formBuilder.control('', Validators.required),
+      },
+      {
+        validator: this.passwordsMatchValidator,
+      }
+    );
+  }
+
+  ngOnInit() {
+    this.activatedRoute.data.subscribe((data: Data) => {
```

Comment:
Do we need to destroy this subscription? You could use https://github.com/ngneat/until-destroy (which we are already using) to deal with that

Reply:
Thanks - Added until destroy as used elsewhere in the code base.